runtime/vm/ContinuationHelpers.cpp

```diff
@@ -91,43 +91,46 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 }
 
 j9object_t
-synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, J9VMContinuation *continuation)
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, ContinuationState volatile *continuationStatePtr)
 {
-	uintptr_t oldContinuationState = 0;
-	uintptr_t returnContinuationState = 0;
+	ContinuationState oldContinuationState = 0;
+	ContinuationState returnContinuationState = 0;
+	/* Set pending state and carrier id */
 	do {
-		oldContinuationState = continuation->state;
-		uintptr_t newContinuationState = oldContinuationState;
+		oldContinuationState = *continuationStatePtr;
+		ContinuationState newContinuationState = oldContinuationState;
 		VM_VMHelpers::settingCarrierAndPendingState(&newContinuationState, currentThread);
-		returnContinuationState = VM_AtomicSupport::lockCompareExchange(&continuation->state, oldContinuationState, newContinuationState);
+		returnContinuationState = VM_AtomicSupport::lockCompareExchange(continuationStatePtr, oldContinuationState, newContinuationState);
 	} while (returnContinuationState != oldContinuationState);
 	Assert_VM_false(VM_VMHelpers::isPendingToBeMounted(returnContinuationState));
 	Assert_VM_Null(VM_VMHelpers::getCarrierThread(returnContinuationState));
 
 	do {
-		if (VM_VMHelpers::isConcurrentlyScanned(returnContinuationState)) {
-			/* currentThread was low tagged (GC was already in progress), but by 'or'-ing our ID, we let GC know there is a pending mount */
-
+		/* Wait till potentially concurrent GC scans are complete */
+		while (VM_VMHelpers::isConcurrentlyScanned(returnContinuationState)) {
```

Comment:
let's add another sentence here:
This loop should iterate only a few times, at worst. We might have up to 2 overlapping concurrent GC, and each of them typically scans the object once. Some GCs (for example card dirtying based) may do rescans, but only up to small finite number of times.

Reply:
actually, this comment applies for both loops, so you can put it as a separate comment ahead of the outter loop. The you update the comment to start with plural 'These loops....'